### ISSUE TITLE
RFC 697: Testing - IAM user story

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#40704]
See [RFC 697: Multiple version upgrades](https://docs.google.com/document/d/1eNiwzWZmfpvzFhs9IDcM7Sbchv_Hm8Uj-6bI-3TDlvg).

This PR tracks general testing of multi-version upgrades as well as IAM-specific test scenarios. 

See the [testing strategy announcement in slack](https://sourcegraph.slack.com/archives/C02KZ941RNX/p1660748283465879).

**Core story (altered for IAM specifics)**:

1. Run a search for the `sourcegraph/sourcegraph` repository
2. Search for the term `GetUploads` on `HEAD`
3. Open a result and ensure the file renders
4. Check that file sidebar and symbols sidebars are populated
5. Ensure code navigation (go to definition and find references) work as expected
6. Repeat steps 2-5 on an old tag to ensure that unindexed paths are also functional
7. Visit each site-admin page to ensure there are no obvious error banners
8. **IAM migration test**: Ensure that account number and license fields were properly migrated

**Instructions**:

1. Start a local instance of Sourcegraph at version 3.32.0. The `docker-compose.yaml` artifact, below, provides an easy way to link a `sourcegraph/server` image with three container-based Postgres instances. The frontend, codeintel, and codeinsights databases are available on the host via ports 7777, 7778, and 7779, respectively.
2. Perform whatever steps listed above that need to be performed prior to upgrading (available on the host via port 7080).
3. Stop the `sourcegraph/server` container, but leave the three database containers running.
4. In a clone of the `sg/sg` repo, run the following commands to upgrade your local instance to version 4.0.

```bash
export PGDATASOURCE=postgres://sg:sg@127.0.0.1:7777/sg
export CODEINTEL_PGDATASOURCE=postgres://sg:sg@127.0.0.1:7778/sg
export CODEINSIGHTS_PGDATASOURCE=postgres://sg:sg@127.0.0.1:7779/sg

go run ./enterprise/cmd/migrator upgrade --from 3.32 --to 4.0
```

5. Update the tag on the `sourcegraph/server` image to point to the "pre-release" of version 4.0. The particular commit that we're treating as the 4.0 release prior to the creation of the tag in git can be found [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/migration/stitch/git.go?L128=). 

**Artifacts**:

`docker-compose.yaml`:

```yaml
version: '2.4'
services:
  sourcegraph:
    image: sourcegraph/server:{{TAG GOES HERE}}
    environment:
      - PGDATASOURCE=postgres://sg:sg@pgsql-frontend:5432/sg
      - CODEINTEL_PGDATASOURCE=postgres://sg:sg@pgsql-codeintel:5432/sg
      - CODEINSIGHTS_PGDATASOURCE=postgres://sg:sg@pgsql-codeinsights:5432/sg
    ports:
      - 7080:7080
    networks:
      - sourcegraph
    depends_on:
      - pgsql-frontend
      - pgsql-codeintel
      - pgsql-codeinsights

  pgsql-frontend:
    image: sourcegraph/postgres-12-alpine
    volumes:
      - 'pgsql-frontend:/data'
    ports:
      - 7777:5432
    networks:
      - sourcegraph

  pgsql-codeintel:
    image: sourcegraph/postgres-12-alpine
    volumes:
      - 'pgsql-codeintel:/data'
    ports:
      - 7778:5432
    networks:
      - sourcegraph

  pgsql-codeinsights:
    image: sourcegraph/postgres-12-alpine
    volumes:
      - 'pgsql-codeinsights:/data'
    ports:
      - 7779:5432
    networks:
      - sourcegraph

volumes:
  sourcegraph:
  pgsql-frontend:
  pgsql-codeintel:
  pgsql-codeinsights:

networks:
  sourcegraph:
```

**Comment by efritz** [comment]
@ryphil Unblocked!

**Comment by pjlast** [comment]
I can confirm that everything worked as expected.

I set up a user with username `petritest-1234` and created a product subscription for the user.

I created a license key attached to the product subscription.

I ran the migration, and in the DB license tags, user count, license expiry and license version was extracted to their own columns, and in the product subscriptions table, the account number was extracted to its own column.
